host.py

```diff
@@ -43,19 +43,23 @@ def add_flow(self, flow):
         self.flows[flow.flow_id] = flow
 
     def send_packet(self, packet):
-        """Sends a packet to another node.
+        """Sends a packet to another node by giving the packet to the host's
+        link.
 
         Args:
             packet (Packet): The packet to send.
 
         """
-        pass
+        event = lambda: self._link.add_packet(packet, self.node_id)
+        ns.add_event(event)
 
     def receive_packet(self, packet):
-        """Receives a packet from another node.
+        """Receives a packet from another node and then tells the flow that the
+        packet came from to receive the packet and respond accordingly.
 
         Args:
             packet (Packet): The packet we received.
             
         """
-        pass
\ No newline at end of file
+        event = lambda: self.flows[packet.flow_id].receive_packet(packet)
```

Comment:
Can you assert packet.flow_id in self.flows here to make sure we don't send packets to the wrong hosts